subsys/lorawan/services/lorawan_services.c

```diff
@@ -160,6 +164,55 @@ int lorawan_services_reschedule_work(struct k_work_delayable *dwork, k_timeout_t
 	return k_work_reschedule_for_queue(&services_workq, dwork, delay);
 }
 
+int lorawan_services_class_c_start(void)
+{
+	int ret;
+
+	k_mutex_lock(&session_mutex, K_FOREVER);
```

Comment:
I think this mutex setup can be achieved with the atomic API.
```
if (atomic_inc(&session_count) == 0) {
  lorawan_set_class(LORAWAN_CLASS_C);
  LOG_DBG("Switched to class C");
}
```
Are there realistic conditions under which `lorawan_set_class(LORAWAN_CLASS_C)` can fail?

Reply:
Yes, it could fail when a transmission is active. Happened to me a couple of times during testing.